### DISCUSSION TITLE
Gradual migration towards parallel tests

### DISCUSSION THREAD

**Discussion by uncaught** [paratestphp/paratest#852] [Q&A] [answered]
I would like to start using paratest, but most of our functional tests are not ready, yet. There is too much static setup setting up test fixtures in the database and between test classes the database gets truncated.

My plan:

- Write new functional tests with DAMA\DoctrineTestBundle enabled, so they never modify the database, which makes them parallelizable
- Leave old tests with DAMA\DoctrineTestBundle disabled and have them still truncate the database
- Migrate old tests over time

My questions:

- Is there a way to tell paratest to run groups of tests in a specific order and with different process numbers?
- In a sense, I'd like paratest to
    - first run all new tests (e.g. "--group parallel") with 4 processes
    - and _then_ run all other tests with just 1 process (effectively not in parallel)

If this is not possible, I'd have to run those two steps manually myself and somehow merge all reports and CI-output together.

But since paratest is already merging results and reports, I could imagine this to be provided as a feature?

My use case of gradually parallelizing a base of tests must surely not be unique.

Or is there maybe another way? Maybe to append to existing reports in successive paratest-runs?

Thanks in advance

**Comment by Slamdunk** [comment, marked as answer]
A lot of code would need to be written to accomplish your goal.

Why not simply create *n* databases, one for each process, as described in https://github.com/paratestphp/paratest?tab=readme-ov-file#test-token?
I've never encountered a project where this is _not_ much cheaper than any other solution.

**Reply by uncaught** [reply]
I intend to use multiple databases, but that doesn't help my goal of getting rid of the truncating (which is a performance killer regardless of parallel or not).

Okay, so if there is no way to control the order of tests, then I guess I'll have to do it manually.

Thanks anyway :)
